src/freenet/client/filter/TheoraPacketFilter.java

```diff
@@ -1,148 +1,287 @@
+		byte[] magicHeader = new byte[1 + magicNumber.length];
+		input.readFully(magicHeader);
+		checkMagicHeader(magicHeader, (byte) 0x80); // -128
+
+		int VMAJ = input.readInt(8);
+		if (VMAJ != 3) throw new UnknownContentTypeException("Header VMAJ: " + VMAJ);
+
+		int VMIN = input.readInt(8);
+		if (VMIN != 2) throw new UnknownContentTypeException("Header VMIN: " + VMIN);
+
+		int VREV = input.readInt(8);
+		if (VREV > 1) throw new UnknownContentTypeException("Header VREV: " + VREV);
+
+		int FMBW = input.readInt(16);
+		if (FMBW == 0) throw new UnknownContentTypeException("Header FMBW: " + FMBW);
+
+		int FMBH = input.readInt(16);
+		if (FMBH == 0) throw new UnknownContentTypeException("Header FMBH: " + FMBH);
+
+		int PICW = input.readInt(24);
+		if (PICW > FMBW * 16) throw new UnknownContentTypeException("Header PICW: " + PICW + "; FMBW: " + FMBW);
+
+		int PICH = input.readInt(24);
+		if (PICH > FMBH * 16) throw new UnknownContentTypeException("Header PICH: " + PICH + "; FMBH: " + FMBH);
+
+		int PICX = input.readInt(8);
+		if (PICX > FMBW * 16 - PICX) throw new UnknownContentTypeException("Header PICX: " + PICX + "; FMBW: " + FMBW + "; PICX: " + PICX);
+
+		int PICY = input.readInt(8);
+		if (PICY > FMBH * 16 - PICY) throw new UnknownContentTypeException("Header PICY: " + PICY + "; FMBH: " + FMBH + "; PICY: " + PICY);
+
+		int FRN = input.readInt(32);
+		if (FRN == 0) throw new UnknownContentTypeException("Header FRN: " + FRN);
+
+		int FRD = input.readInt(32);
+		if (FRD == 0) throw new UnknownContentTypeException("Header FRN: " + FRN);
+
+		input.skip(48); // skip PARN and PARD
+
+		int CS = input.readInt(8);
+		if (!(CS == 0 || CS == 1 || CS == 2)) throw new UnknownContentTypeException("Header CS: " + CS);
+
+		input.skip(35); // skip NOMBR, QUAL and KFGSHIFT
+
+		int PF = input.readInt(2);
+		if (PF == 1) throw new UnknownContentTypeException("Header PF: " + PF);
+
+		int Res = input.readInt(3);
+		if (Res != 0) throw new UnknownContentTypeException("Header Res: " + Res);
+
+		expectedPacket = Packet.COMMENT_HEADER;
+	}
+
+	private CodecPacket parseCommentHeader(BitInputStream input) throws IOException {
+		Logger.minor(this, "COMMENT_HEADER");
+
+		byte[] magicHeader = new byte[1 + magicNumber.length];
+		input.readFully(magicHeader);
+		checkMagicHeader(magicHeader, (byte) 0x81); // -127
+
+		int vendorLength = input.readInt(32, ByteOrder.LITTLE_ENDIAN);
+		byte[] vendor = new byte[vendorLength];
+		input.readFully(vendor);
+		Logger.minor(this, "Vendor string is: " + new String(vendor, StandardCharsets.UTF_8));
+		int numberOfComments = input.readInt(32, ByteOrder.LITTLE_ENDIAN);
+		for (long i = 0; i < numberOfComments; i++) {
+			int commentLength = input.readInt(32, ByteOrder.LITTLE_ENDIAN);
+			byte[] comment = new byte[commentLength];
+			input.readFully(comment);
+			Logger.minor(this, "Comment string is: " + new String(comment));
+		}
+
+		try {
+			input.readBit();
```

Comment:
And the same is valid for this here… if we don’t do anything with the information, what do we get it for? Either redundant bits are an error (in which case we should throw) or redundant bits are totally fine.

Reply:
Spec doesn't say that redundant bits are an error or not. So as we skip comments at next lines of code, let's skip reading them. I removed this fragment.